### PR TITLE
Added Edit Study Plan feature

### LINKED ISSUE #128: [Student View] Edit study plan name feature

### PR DESCRIPTION
**Pull request by NatashaKSS**
* Refactored TabbedContainer, Tab and StudyPlan to break down large React components
* Added code documentation for each of the above
* RESOLVES #128,  #148, #163, #164, #165, #166, #168

### CODE REVIEW

File: imports/ui/components/common/Constants.js [hunk 1]
```diff
@@ -3,3 +3,4 @@ export const SIDEBAR_PANEL_WIDTH = 300;
 export const SIDEBAR_WIDTH = SIDEBAR_MENU_WIDTH + SIDEBAR_PANEL_WIDTH;
 
 export const ENTER_CHAR_KEY_CODE = 13;
+export const MAX_STUDY_PLAN_NAME_LENGTH = 12;
```
Comment (walrys) [inline imports/ui/components/common/Constants.js]: great thinking! can consider limiting the planner name length in the future

File: imports/ui/components/common/TabbedContainer.jsx [hunk 2]
```diff
@@ -1,223 +0,0 @@
...
-
-  // Handler for when a user clicks on delete button on the Tab
-  onClickDeleteTab: React.PropTypes.func,
-
-  // Boolean true means this Tab is currently selected, false otherwise
-  isActiveTab: React.PropTypes.bool,
-
-  // Boolean true means this Tab will deactivate mouse over events, false otherwise
-  enabledMouseOver: React.PropTypes.bool
-}
```
Comment (walrys) [inline imports/ui/components/common/TabbedContainer.jsx]: nice now its easier to read!

File: imports/ui/components/study_plan/Tab.jsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+  // Handler for when a user clicks on delete button on the Tab
+  onClickDeleteTab: React.PropTypes.func,
+
+  // Boolean true means this Tab is currently selected, false otherwise
+  isActiveTab: React.PropTypes.bool,
+
+  // Boolean true means this Tab will deactivate mouse over events, false otherwise
+  enabledMouseOver: React.PropTypes.bool
+}
```
Comment (walrys) [inline imports/ui/components/study_plan/Tab.jsx]: great job with the refactoring. formatting is clear and easy to understand.

File: imports/ui/components/study_plan/TabbedContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+  /**
+   * Renders the title of a tab. If the number of characters for this tab
+   * is too long (>MAX_STUDY_PLAN_NAME_LENGTH chars), it will be sliced so
+   * that tab titles don't contain Strings that are too long to fit in a tab.
+   *
+   * @param {[String]} tabTitle    Text of tab's title
+   * @param {[String]} plannerID     ID of this tab's study plan
+   * @return    Shortened String representation of this tab title.
+   */
```
Comment (walrys) [inline imports/ui/components/study_plan/TabbedContainer.jsx]: Didn't know I can do this in javascript too! will add into future refactoring

File: imports/ui/components/common/Constants.js (hunk 1 above)
Comment (NatashaKSS) [inline imports/ui/components/common/Constants.js]: 👍

File: imports/ui/components/study_plan/TabbedContainer.jsx (hunk 4 above)
Comment (NatashaKSS) [inline imports/ui/components/study_plan/TabbedContainer.jsx]: Yup! Usually @param and @return get highlighted on most text editors (even on GitHub)! So it looks clearer too!

**Pull request merged**